inlang/source-code/paraglide/paraglide-js/src/services/telemetry/stack-detection.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * A list of packages that we consider relevant for stack detection.
+ *
+ * We (Loris & Nils) decided against using _all_ packages because that would be too much unnecessary data
+ * that would slow down queries.
+ */
+const RelevantPackages = [
+	"next",
+	"solid",
+	"solid-start",
+	"svelte",
+	"@sveltejs/kit",
+	"@sveltejs/adaper-static",
+	"@sveltejs/adapter-node",
+	"@sveltejs/adapter-netlify",
+	"@sveltejs/adapter-vercel",
+	"@sveltejs/adapter-cloudflare-workers",
+	"@sveltejs/adapter-auto",
+	"vue",
+	"nuxt",
+	"react",
+	"react-native",
+	"remix",
+	"astro",
+	"flutter",
+	"vite",
+	"webpack",
+	"rollup",
+	"esbuild",
+	"qwick",
+	"parcel",
+	"lit",
+	"lit-html",
+	"@angular/core",
+] as const
+
+type RelevantPackage = (typeof RelevantPackages)[number]
+
+export type StackInfo = {
+	/**
+	 * A map of relevant packages to their versions.
+	 */
+	packages: {
+		[packageName in RelevantPackage]?: string
+	}
+}
+
+export function getStackInfo(packageJson: unknown): StackInfo {
```

Comment:
can you add a comment to the function what information is gathered?

Reply:
Good Point, I'll add a comment. You did guess the correct return type

I nested the packages. because we may want to add further information about the "stack". Having the `packages` be a property would mean we could add that without breaking existing queries in Posthog